gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsBatchResult.java

```diff
@@ -21,44 +21,18 @@
 /**
  * This class holds a single result of a batch call to the Cloud DNS.
  */
-public class DnsBatchResult<T> implements BatchResult<T, DnsException> {
-
-  private T result;
-  private boolean submitted = false;
-  private DnsException error;
+public class DnsBatchResult<T> extends BatchResult<T, DnsException> {
 
   DnsBatchResult() {
   }
 
   @Override
-  public boolean submitted() {
-    return submitted;
-  }
-
-  @Override
-  public T get() throws DnsException {
-    if (!submitted()) {
-      throw new IllegalStateException("Batch has not been submitted yet");
-    }
-    if (error != null) {
-      throw error;
-    }
-    return result;
+  public void error(DnsException error) {
+    super.error(error);
   }
 
   @Override
-  public void notify(Callback<T, DnsException> callback) {
-    // todo(mderka) implement
-    throw new UnsupportedOperationException("Not implemented yet");
-  }
-
-  void error(DnsException error) {
-    this.error = error;
-    this.submitted = true;
-  }
-
-  void success(T result) {
-    this.result = result;
-    this.submitted = true;
+  public void success(T result) {
```

Comment:
Overriding for the method to be callable from this package. However, they can be protected.